client/webserver/site/src/js/forms.js

```diff
@@ -640,27 +644,27 @@ export class WalletWaitForm {
     if (wallet.assetID !== this.assetID) return
     if (this.progressed && this.funded) return
     this.reportProgress(wallet.synced, wallet.syncProgress)
-    this.reportBalance(wallet.balance)
+    this.reportBalance(wallet.balance, wallet.assetID)
   }
 
   /*
    * reportBalance sets the balance display and calls success if we go over the
    * threshold.
    */
-  reportBalance (bal) {
-    if (this.funded || this.assetID === -1) return
+  reportBalance (bal, assetID) {
+    if (this.funded || this.assetID === -1 || this.assetID !== assetID) return
     const page = this.page
     const asset = app().assets[this.assetID]
-    const fee = this.regFee
 
-    if (bal.available <= fee.amount) {
+    if (bal.available <= this.regFee.amount) {
```

Comment:
Feel like it's ok to check the box when `avail == amt` rather than requiring a little more?  I know this doesn't generalize to tokens, but I feel like we'd catch a ton of use mistakes here that would lead to exact deposits and then it failing in `Register` on the backend.

Reply:
The `<=` here leads to an early return statement. It will only check the box if the balance is greater than the registration fee.

Reply:
Ahh, read the flow backwards. Nice, thanks.